storage/client_raft_test.go

```diff
@@ -274,6 +274,23 @@ func TestFailedReplicaChange(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	// Wait for the range to sync to both replicas (mainly so leaktest doesn't
```

Comment:
Yeah, but normally we take care of this by executing commands synchronously. Replication is the only case where we can't do that, and it's rare for us to perform replication in a test without doing something that depends on the result of that replication.